ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/HttpCache.kt

```diff
@@ -151,6 +151,7 @@ public class HttpCache private constructor(
             scope.sendPipeline.intercept(CachePhase) { content ->
                 if (content !is OutgoingContent.NoContent) return@intercept
                 if (context.method != HttpMethod.Get || !context.url.protocol.canStore()) return@intercept
+                if (context.headers.contains(HttpHeaders.Authorization)) return@intercept
```

Comment:
Could this instead be configurable this behaviour as the default?  Assuming the authentication is still enforced for cached responses, it would be useful to cache these responses as well.

Reply:
yep, let me add a config property

Reply:
Hi, @e5l! 😄  A bit late to the game, but the spec only forbids storing of responses to requests with `Authorization` header in _shared_ caches. If we store them in private caches, it is configurable through `isSharedClient` - and then we don't need the `cacheRequestWithAuth` property. I think these kinda things are why there is the distinction between private and shared caches.

Reply:
Hey @NilsBarlaug, thanks for the notice. Let me check this

Reply:
https://github.com/ktorio/ktor/pull/4368